### PR TITLE
charts,images,manifests: Update reporting-operator configuration schema

### PR DESCRIPTION
**Pull request by chancez**
Updates the MeteringConfig spec.reporting-operator section to have more
consistency in configuring options. This is the first of many changes,
where I will be updating all components to have consistency across
components and consistency within components.

### CODE REVIEW

File: charts/openshift-metering/templates/reporting-operator/reporting-operator-deployment.yaml [hunk 1]
```diff
@@ -365,60 +372,62 @@ spec:
...
           name: cookie-secret
         - mountPath: /etc/proxy/htpasswd
           name: htpasswd-secret
-{{- if $operatorValues.spec.authProxy.authenticatedEmailsEnabled  }}
+{{- if $operatorValues.spec.authProxy.authenticatedEmails.enabled  }}
         - mountPath: /etc/proxy/authenticated-emails
           name: authenticated-emails-secret
 {{- end }}
-{{- end }}
+{{- end }}{{/* end of authProxy.enabled */}}
```
Comment (timflannagan1) [inline charts/openshift-metering/templates/reporting-operator/reporting-operator-deployment.yaml]: nice, I had a hard time following this when I was initially looking at the reporting operator deployment.

File: charts/openshift-metering/values.yaml [hunk 2]
```diff
@@ -212,146 +212,206 @@ reporting-operator:
...
-      disablePrometheusMetricsImporter: "false"
-      enableFinalizers: "false"
-      hiveHost: hive-server:10000
-      leaderLeaseDuration: 60s
-      logDDLQueries: "false"
-      logDMLQueries: "false"
+      enableFinalizers: false
+      leaderLeaseDuration: "60s"
+
+      logDDLQueries: false
```
Comment (timflannagan1) [inline charts/openshift-metering/values.yaml]: I still prefer having a `logging` field that would store these, but if you think unnecessary, that's fine with me.

Comment (chancez) [reply charts/openshift-metering/values.yaml]: I just feel like most components may not have many logging options, and they're all kind app specific so it seems generalizing this isn't as useful.

**Pull request merged**